### PR TITLE
Allow negative index for nvim_buf_set_lines

### PR DESCRIPTION
**Pull request by tbo**
This PR adds support for negative indices in `Buffer.setLines`.

Excerpt from the `nvim_buf_set_lines` documentation:

_"Indexing is zero-based, end-exclusive. Negative indices are interpreted as length+1+index: -1 refers to the index past the end. So to change or delete the last element use start=-2 and end=-1."_

Example usage:
```
buffers[0].set_lines(&mut nvim, -1, -1, true, vec!["append to buffer".to_owned()]).unwrap();
```

### CODE REVIEW

**Comment by daa84** [comment]
Problem is that neovim_api.rs is generated from template with python script.
Currently in this script Integer type binded to u64, maybe better to update script and regenerate all bindings, so that it always maps to i64? Think u64 was a mistake. 
script and template are at bindings directory.

**Comment by KillTheMule** [comment]
Are you dead set on only using that script? You could autogenerate the API via the script and do some manual transformations. There's a cargo plugin that warns you about semver-incompatible changes, that should catch if you forget to change u64 -> i64 somewhere.

File: src/rpc/model.rs [hunk 1]
```diff
@@ -8,12 +8,12 @@ use std::io::{Read, Write};
 #[derive(Debug, PartialEq, Clone)]
 pub enum RpcMessage {
     RpcRequest {
-        msgid: u64,
+        msgid: i64,
```
Comment (daa84) [inline src/rpc/model.rs]: think here u64 is ok

File: src/rpc/model.rs [hunk 2]
```diff
@@ -8,12 +8,12 @@ use std::io::{Read, Write};
...
         method: String,
         params: Vec<Value>,
     }, // 0
     RpcResponse {
-        msgid: u64,
+        msgid: i64,
```
Comment (daa84) [inline src/rpc/model.rs]: and here 😄

**Comment by daa84** [comment]
@KillTheMule i'd prefer not do manual changes. In case it is need manual tweak it is better to extract function to other place and skip it in generator. 
But think Integer == i64 in our case, so think u64 was bad solution

**Comment by daa84** [comment]
Thank you

**Pull request merged**